File: src/duplicate_array_descriptors.c
#include <stdio.h>
#include <stdlib.h>
int main(void) {
	const int length = 1024*1024;
	double * arr = (double *) malloc(length*sizeof(double));
	double tmp;
	#pragma omp parallel for private(tmp) 
	for(int i = 0; i < length; i++) {
		arr[i] = 1.0/length;
		tmp= arr[i];
	}
	printf("The result of arr[0]+arr[%d] is %1.9lf\n",length-1,arr[0]-arr[length-1]);
	free(arr);
	return 0;
}

File: src/duplicate_array_descriptors.f90
PROGRAM duplicate_array_descriptors
!-----------------------------------------------------------------------
!  Minimal reproducible example of flang-new generating duplicate array 
! descriptors with -fopenmp -O3
!-----------------------------------------------------------------------
IMPLICIT NONE
! Declare local variables
INTEGER(kind=4)			:: length, i
REAL(kind=8), allocatable 	:: arr(:)
REAL(kind=8)			:: tmp
length = 1024*1024
allocate (arr(length))
! Parallel do loop
!$omp parallel do private(tmp)
do i=1,length
	arr(i) = 1.0/length
	tmp = arr(i)
!	arr(i) = 2.0*arr(i)
!	arr(i) = 0.5*arr(i)
end do
!$omp end parallel do
! Write the result
write(*,100) "The result of (arr(1)+arr(",length,") is ", (arr(1)+arr(length))
100 format (A,I7,A,e13.6e2)
! Deallocate Array
deallocate(arr)
END PROGRAM duplicate_array_descriptors
